benchmarks/statevec.py

```diff
@@ -0,0 +1,208 @@
+"""
+Statevector simulation of MBQC patterns
+=======================================
+
+Here we benchmark our statevector simulator for MBQC.
+
+The methods and modules we use are the followings:
+    1. :meth:`~graphix.pattern.Pattern.simulate_pattern`
+        Pattern simulator with statevector backend.
+    2. :mod:`paddle_quantum.mbqc`
+        Pattern simulation using :mod:`paddle_quantum.mbqc`.
+    3. :mod:`mentpy`
+        Pattern simulation using :mod:`mentpy`.
+"""
+
+# %%
+# Firstly, let us import relevant modules:
+
+from time import perf_counter
+import matplotlib.pyplot as plt
+import numpy as np
+from graphix import Circuit
+from graphix.clifford import CLIFFORD_MUL
+from graphix.sim.statevec import Statevec, StatevectorBackend, meas_op
+from paddle import to_tensor  # TODO: in macOS, this line might cause error
+from paddle_quantum.mbqc.qobject import Circuit as PaddleCircuit
+from paddle_quantum.mbqc.transpiler import transpile as PaddleTranspile
+from paddle_quantum.mbqc.simulator import MBQC as PaddleMBQC
+import mentpy as mp
+
+# %%
+# Next, we define the simulation runner:
+
+
+def run(backend):
+    """Perform MBQC simulation"""
+    for cmd in backend.pattern.seq:
+        if cmd[0] == "N":
+            backend.add_nodes([cmd[1]])
+        elif cmd[0] == "E":
+            backend.entangle_nodes(cmd[1])
+        elif cmd[0] == "M":
+            backend.measure(cmd)
+        elif cmd[0] == "X":
+            backend.correct_byproduct(cmd)
+        elif cmd[0] == "Z":
+            backend.correct_byproduct(cmd)
+        elif cmd[0] == "C":
+            backend.apply_clifford(cmd)
+        else:
+            raise ValueError("invalid commands")
+        if backend.pattern.seq[-1] == cmd:
+            backend.finalize()
```

Comment:
why not use `pattern.simulate_pattern()` as stated above? perhaps this is for 'old ptrace method' which you had below (I removed this, because there's no point benchmarking our old, inefficient code)